geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxyTest.java

```diff
@@ -109,23 +122,59 @@ public void deliverMessageWhenSubjectIsNotNull() {
   public void deliverMessageWhenSubjectIsNull() {
     when(proxyStatsFactory.create(any(), any(), any()))
         .thenReturn(mock(CacheClientProxyStats.class));
-    proxy = spy(new CacheClientProxy(cache, notifier, socket, id, true, (byte) 1, version, 1L, true,
-        securityService, null, clock, statsFactory, proxyStatsFactory, dispatcherFactory));
-    assertThat(proxy.getSubject()).isNull();
+    proxyWithMultiUser =
+        new CacheClientProxy(cache, notifier, socket, id, true, (byte) 1, version, 1L, true,
+            securityService, null, clock, statsFactory, proxyStatsFactory, dispatcherFactory,
+            clientUserAuths);
+    assertThat(proxyWithMultiUser.getSubject()).isNull();
     Conflatable message = mock(ClientUpdateMessage.class);
     when(securityService.needPostProcess()).thenReturn(true);
     when(proxyStatsFactory.create(any(), any(), any()))
         .thenReturn(mock(CacheClientProxyStats.class));
-    proxy.deliverMessage(message);
+    proxyWithMultiUser.deliverMessage(message);
     verify(securityService, never()).bindSubject(subject);
     verify(securityService, never()).postProcess(any(), any(), any(), anyBoolean());
   }
 
   @Test
   public void replacingSubjectShouldNotLogout() {
-    proxy = new CacheClientProxy(cache, notifier, socket, id, true, (byte) 1, version, 1L, true,
-        securityService, subject, clock, statsFactory, proxyStatsFactory, dispatcherFactory);
-    proxy.setSubject(mock(Subject.class));
+    proxyWithSingleUser.setSubject(mock(Subject.class));
+    verify(subject, never()).logout();
+  }
+
+  @Test
+  public void close_keepProxy_ShouldNotLogoutUser() {
+    when(id.isDurable()).thenReturn(true);
+    boolean keepProxy = proxyWithSingleUser.close(true, false);
+    assertThat(keepProxy).isTrue();
+    verify(subject, never()).logout();
+    verify(clientUserAuths, never()).cleanup(anyBoolean());
+
+    keepProxy = proxyWithMultiUser.close(true, false);
+    assertThat(keepProxy).isTrue();
+    verify(subject, never()).logout();
+    verify(clientUserAuths, never()).cleanup(anyBoolean());
+  }
+
+  @Test
+  public void close_singleUser() {
```

Comment:
Could this test name be a little more descriptive? Adding what the expected behaviour is would make it much clearer what the test is actually testing. This also applies to the `close_multiUser()` test.